packages/perseus-editor/src/components/__tests__/start-coord-settings.test.tsx

```diff
@@ -152,4 +156,159 @@ describe("StartCoordSettings", () => {
+    });
+
+    describe.each`
+        type
+        ${"linear-system"}
+        ${"segment"}
+    `(`graphs with CollinearTuple[] startCoords`, ({type}) => {
+        const multilineProps = {
+            ...defaultProps,
+            type,
+            onChange: () => {},
+        };
+        const segmentProps = {
+            ...multilineProps,
+            numSegments: 2,
+        };
+        it("shows the start coordinates UI for 2 segments", () => {
+            // Arrange
+
+            // Act
+            render(
+                <StartCoordSettings
+                    {...(type === "linear-system"
+                        ? multilineProps
+                        : segmentProps)}
+                />,
+                {wrapper: RenderStateRoot},
+            );
+
+            // Assert
+            expect(screen.getByText("Start coordinates")).toBeInTheDocument();
+            expect(screen.getByText("Segment 1")).toBeInTheDocument();
+            expect(screen.getByText("Segment 2")).toBeInTheDocument();
+            expect(screen.getAllByText("Point 1")).toHaveLength(2);
+            expect(screen.getAllByText("Point 2")).toHaveLength(2);
+        });
+
+        it.each`
+            segmentIndex | pointIndex | coordIndex
+            ${0}         | ${0}       | ${0}
+            ${0}         | ${0}       | ${1}
+            ${0}         | ${1}       | ${0}
+            ${0}         | ${1}       | ${1}
+            ${1}         | ${0}       | ${0}
+            ${1}         | ${0}       | ${1}
+            ${1}         | ${1}       | ${0}
+            ${1}         | ${1}       | ${1}
+        `(
+            `calls onChange when $coord coord is changed (segment $segmentIndex)`,
+            async ({segmentIndex, pointIndex, coordIndex}) => {
+                // Arrange
+                const onChangeMock = jest.fn();
+
+                const coords = [
+                    [
+                        [1, 1],
+                        [2, 2],
+                    ],
+                    [
+                        [3, 3],
+                        [4, 4],
+                    ],
+                ] satisfies CollinearTuple[];
+
+                // Act
+                render(
+                    <StartCoordSettings
+                        {...(type === "linear-system"
+                            ? multilineProps
+                            : segmentProps)}
+                        onChange={onChangeMock}
+                        startCoords={coords}
+                    />,
+                    {wrapper: RenderStateRoot},
+                );
+
+                // Assert
+                const input = screen.getAllByRole("spinbutton", {
+                    name: /coord/,
+                })[segmentIndex * 4 + pointIndex * 2 + coordIndex];
+                await userEvent.clear(input);
+                await userEvent.type(input, "101");
+
+                const expectedCoords = coords;
```

Comment:
It looks like the that's in the `perseus` repo, whereas I'm working in the `perseus-editor` repo. Would it make sense to export that via `index.ts`?

Reply:
Ooh, interesting. Perhaps we should move that clone function to `perseus-core`.

Another alternative is to just copy the `clone` function to `perseus-editor`. "A little copying is better than a little dependency." —[Rob Pike (?)](https://go-proverbs.github.io/)

Reply:
Wait! TIL there's a [built in deep clone](https://www.builder.io/blog/structured-clone)? It seem to work!

![Screenshot 2024-07-24 at 2 59 36 PM](https://github.com/user-attachments/assets/0552ef04-38d2-4e4e-a80c-5561774b5fa9)

Reply:
That's a recent feature... and [it looks like Safari might once again be the problem here](https://caniuse.com/?search=structuredClone). We have to support Safari 15, which includes 15.3, which doesn't have structuredClone.

Can you create a ticket so we remember to switch to structuredClone once it's safe?

Reply:
https://khanacademy.atlassian.net/browse/LEMS-2203

Reply:
Thank you!

(Also, oops! I deleted my comment about `structuredClone` because it didn't seem to work when I tried it and ran tests, but I didn't see you had added comments about it until I refreshed this page.)